src/Features/Core/Portable/UnusedReferences/IReferenceCleanupService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.UnusedReferences
+{
+    internal interface IReferenceCleanupService : IWorkspaceService
+    {
+        /// <summary>
+        /// Gets the current selected TargetFramework for the specified project.
+        /// </summary>
+        string GetTargetFramworkMoniker(ProjectId projectId);
```

Comment:
Framwork

Reply:
is there a reason we need to know the current selecte tfm, as opposed to all the tfms the project supports?

Reply:
should this be async?